### PR TITLE
libtcmu: add support for write protect

### PR DESCRIPTION
**Pull request by liulanzheng**
As discussed in #654, add an api to set write-protect for dev, and set the corresponding parameter when mode sense.

Signed-off-by: liulanzheng <lanzheng.liulz@alibaba-inc.com>

### CODE REVIEW

**Comment by lxbsz** [comment]
Do you have any use case for this ?

**Comment by liulanzheng** [comment]
@lxbsz  in [overlaybd](https://github.com/containerd/overlaybd), we output container images as block device through libtcmu for data acceleration. Images, like overlaybd containerd images, and vm images, also iso images, are readonly, it is better to provided a method to mark the device as readonly fundamentally, which is significantly better than executing blockdev --setro command on client. 
At first, we use tgtd as the backstore and it has readonly option for target configuration. So i think adding an option for write-protect/readonly is a real need.

**Comment by lxbsz** [comment]
BTW, which spc or sbc you are referring to for `orig_buf[3]` and `orig_buf[2]` ? From the sbc for MODE_SENSE_10 the first 8 bytes are reserved only for the `NUMBER OF LOGICAL BLOCKS`, and for MODE_SENSE there are 4 bytes.

**Comment by lxbsz** [comment]
> BTW, which spc or sbc you are referring to for `orig_buf[3]` and `orig_buf[2]` ? From the sbc for MODE_SENSE_10 the first 8 bytes are reserved only for the `NUMBER OF LOGICAL BLOCKS`, and for MODE_SENSE there are 4 bytes.

Okay, I misread the docs and I see it.

File: libtcmu.c [hunk 1]
```diff
@@ -1021,6 +1021,16 @@ bool tcmu_dev_get_unmap_enabled(struct tcmu_device *dev)
 	return dev->unmap_enabled;
 }
 
+bool tcmu_dev_set_write_protect_enabled(struct tcmu_device *dev, bool enabled)
```
[the remark this answers is not in the document]
Comment (liulanzheng) [reply libtcmu.c]: sorry, i remember i fixed it, but ...   it's fixed now.

File: libtcmu_priv.h [hunk 2]
```diff
@@ -55,6 +55,7 @@ struct tcmu_device {
 	unsigned int write_cache_enabled:1;
 	unsigned int solid_state_media:1;
 	unsigned int unmap_enabled:1;
+	unsigned int write_protect_enabled: 1;
```
Comment (lxbsz) [inline libtcmu_priv.h:58]: Please remove the white space before `1`.

Comment (liulanzheng) [reply libtcmu_priv.h]: done.

**Review by lxbsz — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**